src/scenic/syntax/veneer.py

```diff
@@ -441,8 +438,6 @@ def makeRequirement(ty, reqID, req, line, name):
 		raise RuntimeParseError(f'tried to use "{ty.value}" inside a requirement')
 	elif currentBehavior is not None:
 		raise RuntimeParseError(f'"{ty.value}" inside a behavior on line {line}')
-	elif currentSimulation is not None:
-		currentScenario._addDynamicRequirement(ty, req, line, name)
```

Comment:
Removing this case looks suspect to me: since `_addRequirement` just adds the requirement to a list of pending requirements, won't the requirement not get enforced?

Reply:
You're right that these two lines are required and I've restored the changes in 85be02d90ee932129e77933b329a1bcbf58e676d.

I deleted them because no test case uses this branch, so I added one in 3aaf090547648b25d14884c3833f91bda694cc3b

Also, there was an assertion that was no longer valid in `dynamics.py` so I deleted it. 7e00a84c44b73c3ed6cdb97ea380d145e0b49a79

Before this PR, `ty` of `_addDynamicRequirement` needed to be anything but `require` because plain requirements are not dynamic requirements. 

https://github.com/BerkeleyLearnVerify/Scenic/blob/9215576521c1c533d0f71de17d15e706fe31a991/src/scenic/core/dynamics.py#L490

With temporal requirements, plain requirements can also be passed as a dynamic requirement. Therefore, I'm removing this assertion.